mavros_extras/src/plugins/distance_sensor.cpp

```diff
@@ -222,17 +222,14 @@ class DistanceSensorPlugin : public MavRosPlugin {
 
 		if (sensor->send_tf) {
 			/* variables init */
-			// XXX #319
 			auto rpy = UAS::sensor_orientation_matching(static_cast<MAV_SENSOR_ORIENTATION>(dist_sen.orientation));
-			// how it can work, if rpy in degrees, not radians?
 			auto q = tf::createQuaternionFromRPY(rpy.x(), rpy.y(), rpy.z());
```

Comment:
And quaternion are used in ros for all orientation/rotation things. I don't see any place where you need euler angles now. But i see where quaternion is usable.
Also to do rotation you steel need make quaternion and then apply it to Affine3d or other Q.

So no reason why store this data uncalculated and do calc at each call. That data is a constant, so just do precalculation.